src/hooks/useSettings.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from "react";
+
+import { DEFAULT_SHOW_CHAT } from "settings";
+
+import { Settings } from "types/Firestore";
+import { SparkleSelector } from "types/SparkleSelector";
+
+import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+export const settingsSelector: SparkleSelector<Settings | undefined> = (
+  state
+) => state.firestore.data.settings;
+
+export const useSettings: () => Required<Settings> = () => {
+  useFirestoreConnect([
+    { collection: "settings", doc: "customizations", storeAs: "settings" },
+  ]);
+
+  const settings = useSelector(settingsSelector);
+
+  return useMemo(
+    () => ({
+      showChat: settings?.showChat ?? DEFAULT_SHOW_CHAT,
+    }),
+    [settings]
```

Comment:
Few notes on this. Nothing mandatory.

1. Construction can be done in order:

```ts
()=>({
  ...DEFAULT_VALUES_FIRST,
  ...valuesFromDbThatExistWillOverwrite
})
```

2. The above, however, doesn't guard much with DB types being incorrect like getting a string when array or boolean is expected, but I guess for now a more generic solution might be OOS.
For the specific property `settings?.showChat` can be checked with `=== true` or `=== false` before applying 
the default, or just converted with `!!`.

3. Will this PR be followed up with another that adds setting for reactions as well?